packages/angular_devkit/build_ng_packagr/src/build/index.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { BuildEvent, Builder, BuilderContext, Target } from '@angular-devkit/architect';
+import { getSystemPath } from '@angular-devkit/core';
+import * as ngPackagr from 'ng-packagr';
+import { resolve as resolvePath } from 'path';
+import { Observable } from 'rxjs/Observable';
+
+// XX: blatantly copy-pasted from 'require-project-module.ts'
+const resolve = require('resolve');
+function requireProjectModule(root: string, moduleName: string) {
+  return require(resolve.sync(moduleName, { basedir: root }));
+}
+
+
+export interface NgPackagrBuilderOptions {
+  project: string;
+}
+
+export class NgPackagrBuilder implements Builder<NgPackagrBuilderOptions> {
+
+  constructor(public context: BuilderContext) { }
+
+  run(target: Target<NgPackagrBuilderOptions>): Observable<BuildEvent> {
+    const root = getSystemPath(target.root);
+    const options = target.options;
+
+    if (!options.project) {
+      throw new Error('A "project" must be specified to build a library\'s npm package.');
+    }
+
+    return new Observable(obs => {
+      const projectNgPackagr = requireProjectModule(root, 'ng-packagr') as typeof ngPackagr;
+      const packageJsonPath = resolvePath(root, options.project);
+
+      projectNgPackagr.ngPackagr()
+        .forProject(packageJsonPath)
+        .build()
+        .then(() => obs.complete())
```

Comment:
Overall we're trying to go for a model where builders always emit events. This wasn't the case before when you started working on it but since then we revised that because otherwise it was harder to test and reason about it.

The general idea is:
- whenever a builder finished a build, it will emit `{success: true}` if it succeeded or `{success: false}` if it did not
- when a builder is done, it will complete. Builders on watch mode would emit multiple times and not complete, whereas single run builders should emit once and complete.
- when there is an error, the builder should `obs.error`. Note that this is different from a failed build though, which emits a `{success: false}` and no error. I think the usage of error you have here is the right one.